### PR TITLE
docs(cn): translate src/content/api/plugins.md

### PR DESCRIPTION
**Pull request by chenzesam**
_describe your changes..._

- [ ] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [ ] Make sure your PR complies with the [writer's guide][2].
- [ ] Review the diff carefully as sometimes this can reveal issues.
- [ ] Do not abandon your Pull Request: [Stale Pull Requests][3].
- __Remove these instructions from your PR as they are for your eyes only.__

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

**Comment by chenzesam** [comment]
@QC-L Come and review

File: src/content/api/plugins.md [hunk 1]
```diff
@@ -11,102 +11,76 @@ contributors:
   - wizardofhogwarts
 ---
 
-Plugins are a key piece of the webpack ecosystem and provide the community with
-a powerful way to tap into webpack's compilation process. A plugin is able to
-[hook](/api/compiler-hooks/#hooks) into key events that are fired throughout each compilation. Every step
-of the way, the plugin will have full access to the `compiler` and, when
-applicable, the current `compilation`.
+插件是 webpack 生态的关键部分，它为社区用户提供了一种强有力的方式来直接触及 webpack 的编译过程(compilation process)。插件能够[钩入(hook)](/api/compiler-hooks/#hooks)到每一个编译(compilation)中发出的关键事件中。在编译的每一步中，插件都拥有对 `compiler` 对象的完全访问能力，并且在合适的时候还可以访问当前的 `compilation` 对象。
```
Comment (QC-L) [inline src/content/api/plugins.md:14]: 以 ，号断句尽可能的保证行数一致

Comment (QC-L) [reply src/content/api/plugins.md:14]: 在 [Files Change](https://github.com/docschina/webpack.js.org/pull/759/files) 没有一一对应，在之后的同步过程中，会出现意想不到的问题

Comment (QC-L) [reply src/content/api/plugins.md:14]: 后续内容，与此处相同

Comment (chenzesam) [reply src/content/api/plugins.md]: 好了，对齐了~

Comment (QC-L) [reply src/content/api/plugins.md]: @chenzesam 👍

File: src/content/api/plugins.md [hunk 2]
```diff
@@ -11,102 +11,102 @@ contributors:
...
 
-T> For a high-level introduction to writing plugins, start with
-[writing a plugin](/contribute/writing-a-plugin).
+T > 关于编写插件的高级介绍，请移步：
+[编写一个插件](/contribute/writing-a-plugin)。
 
-Let's start by going over `tapable` utility, which provides the backbone of
-webpack's plugin interface.
+让我们首先从 tapable 工具开始，
+它为 webpack 插件接口提供了核心能力的 。
```
Comment (QC-L) [inline src/content/api/plugins.md:24]:
```suggestion
它为 webpack 插件接口提供了核心能力的。
```

Comment (chenzesam) [reply src/content/api/plugins.md]: 合并进去了

File: src/content/api/plugins.md [hunk 3]
```diff
@@ -11,102 +11,102 @@ contributors:
...
 ---
 
-Plugins are a key piece of the webpack ecosystem and provide the community with
-a powerful way to tap into webpack's compilation process. A plugin is able to
-[hook](/api/compiler-hooks/#hooks) into key events that are fired throughout each compilation. Every step
-of the way, the plugin will have full access to the `compiler` and, when
-applicable, the current `compilation`.
+插件是 webpack 生态的关键部分，
+它为社区用户提供了一种强有力的方式来直接触及 webpack 的编译过程(compilation process)。
+插件能够[钩入(hook)](/api/compiler-hooks/#hooks)到每一个编译(compilation)中发出的关键事件中。
```
Comment (QC-L) [inline src/content/api/plugins.md:16]:
```suggestion
插件能够 [hook](/api/compiler-hooks/#hooks) 每个编译阶段触发的关键事件。
```

Comment (QC-L) [reply src/content/api/plugins.md:16]: 感觉这样更通顺一些

File: src/content/api/plugins.md [hunk 4]
```diff
@@ -11,102 +11,102 @@ contributors:
...
+在编译的每一步中，插件都拥有对 `compiler` 对象的完全访问能力，
```
Comment (QC-L) [inline src/content/api/plugins.md:17]:
```suggestion
在编译的每个阶段中，插件都拥有对 `compiler` 对象的完全访问能力，
```

File: src/content/api/plugins.md [hunk 5]
```diff
@@ -11,102 +11,102 @@ contributors:
...
+并且在合适的时候还可以访问当前的 `compilation` 对象。
```
Comment (QC-L) [inline src/content/api/plugins.md:18]:
```suggestion
并且在合适的时机，还可以访问当前的 `compilation` 对象。
```

File: src/content/api/plugins.md [hunk 6]
```diff
@@ -11,102 +11,102 @@ contributors:
...
 
-T> For a high-level introduction to writing plugins, start with
-[writing a plugin](/contribute/writing-a-plugin).
+T > 关于编写插件的高级介绍，请移步：
```
Comment (QC-L) [inline src/content/api/plugins.md:20]:
```suggestion
T> 关于编写插件的高级介绍，请移步：
```

File: src/content/api/plugins.md [hunk 7]
```diff
@@ -11,102 +11,102 @@ contributors:
...
+[编写一个插件](/contribute/writing-a-plugin)。
```
Comment (QC-L) [inline src/content/api/plugins.md:21]:
```suggestion
[自定义插件](/contribute/writing-a-plugin)。
```

File: src/content/api/plugins.md (hunk 3 above)
Comment (QC-L) [reply src/content/api/plugins.md:16]: 通篇修改下，注意中英文间排版。

Comment (chenzesam) [reply src/content/api/plugins.md:16]: 感觉不是专有名词，所以就翻了，不过神韵可能就不见了~

Comment (chenzesam) [reply src/content/api/plugins.md:16]: > 通篇修改下，注意中英文间排版。

好的。

File: src/content/api/plugins.md [hunk 8]
```diff
@@ -11,102 +11,102 @@ contributors:
...
 
-This small library is a core utility in webpack but can also be used elsewhere
-to provide a similar plugin interface. Many objects in webpack extend the
-`Tapable` class. The class exposes `tap`, `tapAsync`, and `tapPromise` methods
-which plugins can use to inject custom build steps that will be fired
-throughout a compilation.
+这个小型库是 webpack 的一个核心工具，但也可用于其他地方，
+以提供类似的插件接口。
+在 webpack 中的许多对象都扩展自 `Tapable` 类。
+它对外暴露了 `tap`，`tapAsync`, 和 `tapPromise` 方法，
```
Comment (QC-L) [inline src/content/api/plugins.md:32]:
```suggestion
它对外暴露了 `tap`，`tapAsync` 和 `tapPromise` 等方法，
```

Comment (QC-L) [reply src/content/api/plugins.md:32]: 多了个 , ，原文也多了

File: src/content/api/plugins.md (hunk 3 above)
Comment (QC-L) [reply src/content/api/plugins.md:16]: 因为钩入读着很奇怪，之前我们讨论过，就不如直接保留

File: src/content/api/plugins.md [hunk 9]
```diff
@@ -11,102 +11,102 @@ contributors:
...
+插件可以使用这些方法注入到自定义的构建步骤中，这些步骤将在整个编译过程的不同时机触发。
 
-Please see the [documentation](https://github.com/webpack/tapable) to learn
-more. An understanding of the three `tap` methods, as well as the hooks that
-provide them, is crucial. The objects that extend `Tapable` (e.g. the compiler),
-the hooks they provide, and each hook's type (e.g. the `SyncHook`) will be
-noted.
+请查看[文档](https://github.com/webpack/tapable)了解更多知识。
```
Comment (QC-L) [inline src/content/api/plugins.md:35]:
```suggestion
请查阅[文档](https://github.com/webpack/tapable)了解更多知识。
```

File: src/content/api/plugins.md [hunk 10]
```diff
@@ -11,102 +11,102 @@ contributors:
...
+理解上面的的三种 `tap` 方法，
+以及提供这些方法的钩子(hooks)对于编写插件来说是至关重要的。
+那些扩展自 `Tapable` 的对象（例如：compiler），
+以及其提供的钩子(hooks)和每个钩子的类型（例如：`同步钩子(SyncHook)`）是值得我们注意的。
```
Comment (QC-L) [inline src/content/api/plugins.md:39]:
```suggestion
以及其提供的钩子(hooks)和每个钩子的类型（例如：`同步钩子(SyncHook)`）值得关注。
```

File: src/content/api/plugins.md [hunk 11]
```diff
@@ -11,102 +11,102 @@ contributors:
...
 ``` js
 compiler.hooks.compile.tap('MyPlugin', params => {
-  console.log('Synchronously tapping the compile hook.');
+  console.log('以同步方式触及 compile 钩子。');
 });
 ```
 
-However, for `run` which utilizes the `AsyncHook`, we can utilize `tapAsync`
-or `tapPromise` (as well as `tap`):
+然而，对于可以使用 `异步钩子(AsyncHook)` 的 `运行(run)` 阶段，
```
Comment (QC-L) [inline src/content/api/plugins.md:60]:
```suggestion
然而，对于可以使用 `AsyncHook` 的 `run` 阶段，
```

File: src/content/api/plugins.md [hunk 12]
```diff
@@ -11,102 +11,102 @@ contributors:
...
+我们能够使用 `tapAsync` 或 `tapPromise`（以及 `tap`）方法。
```
Comment (QC-L) [inline src/content/api/plugins.md:61]:
```suggestion
则需使用 `tapAsync` 或 `tapPromise`（以及 `tap`）方法。
```

File: src/content/api/plugins.md [hunk 13]
```diff
@@ -11,102 +11,102 @@ contributors:
...
-This small library is a core utility in webpack but can also be used elsewhere
-to provide a similar plugin interface. Many objects in webpack extend the
-`Tapable` class. The class exposes `tap`, `tapAsync`, and `tapPromise` methods
-which plugins can use to inject custom build steps that will be fired
-throughout a compilation.
+这个小型库是 webpack 的一个核心工具，但也可用于其他地方，
+以提供类似的插件接口。
+在 webpack 中的许多对象都扩展自 `Tapable` 类。
+它对外暴露了 `tap`，`tapAsync`, 和 `tapPromise` 方法，
+插件可以使用这些方法注入到自定义的构建步骤中，这些步骤将在整个编译过程的不同时机触发。
```
Comment (QC-L) [inline src/content/api/plugins.md:33]: @hijiangtao 这段，没想好怎么翻，瞅瞅

File: src/content/api/plugins.md [hunk 14]
```diff
@@ -11,102 +11,102 @@ contributors:
...
+这些需求(story)的含义在于，
+我们可以有多种方式钩入(hook)到 compiler 中，可以让各种插件都以合适的方式去运行。
 
 
-## Custom Hooks
+## 自定义钩子
 
-In order to add a new hook to the compilation for other plugins to `tap` into,
-simply `require` the necessary hook class from `tapable` and create one:
+如果我们要让其他插件在编译过程中可以`触及(tap)`而创建一个新的钩子(hook)，
```
Comment (QC-L) [inline src/content/api/plugins.md:87]: tap 就别翻译了，是 Tapable 提供的方法

Comment (QC-L) [inline src/content/api/plugins.md:87]:
```suggestion
为了便于其他插件的编译过程中可以 `tap` 到，则需要创建一个新的 hook
```

File: src/content/api/plugins.md [hunk 15]
```diff
@@ -11,102 +11,102 @@ contributors:
...
+我们只需要简单的从 `tapable` 中 `require` 所需的钩子类(hook class)，然后创建：
```
Comment (QC-L) [inline src/content/api/plugins.md:88]:
```suggestion
我们只需要简单的从 `tapable` 中 `require` 所需的 hook 类，并创建：
```

File: src/content/api/plugins.md [hunk 16]
```diff
@@ -122,26 +122,26 @@ compiler.hooks.emit.tapAsync({
...
-- `percentage`: This argument is unused; instead, [`ProgressPlugin`](/plugins/progress-plugin/) will calculate a percentage based on the current hook.
-- `...args`: Any number of strings, which will be passed to the `ProgressPlugin` handler to be reported to the user.
+- `percentage`：此参数未使用。作为代替，[`ProgressPlugin`](/plugins/progress-plugin/) 插件会基于当前的钩子(hook)计算进度。
+- `...args`：任意数量的字符串，这些字符串会传递给 `ProgressPlugin` 插件并报告给用户。
 
-Note that only a subset of compiler and compilation hooks support the `reportProgress` function. See [`ProgressPlugin`](/plugins/progress-plugin/#supported-hooks) for a full list.
+注意：只有 compiler 和 compilation 钩子的子集才支持 `reportProgress` 方法。请查看 [`ProgressPlugin`](/plugins/progress-plugin/#supported-hooks) 了解更多信息。
 
-## Logging
+## 记录
```
Comment (QC-L) [inline src/content/api/plugins.md:136]:
```suggestion
## 日志
```

File: src/content/api/plugins.md [hunk 17]
```diff
@@ -122,26 +122,26 @@ compiler.hooks.emit.tapAsync({
...
 
-Logging API is available since the release of webpack 4.37. When `logging` is enabled in [`stats configuration`](/configuration/stats/#statslogging) and/or when [`infrastructure logging`](/configuration/other-options/#infrastructurelogging) is enabled, plugins may log messages which will be printed out in the respective logger format (stats, infrastructure).
+记录(Logging)接口在 webpack 4.37 版本后提供支持。当 `logging` 在 [`统计配置(stats configuration)`](/configuration/stats/#statslogging)中可用和(或)当 [`infrastructure logging`](/configuration/other-options/#infrastructurelogging) 可用的时候，插件会通过各自的记录格式(stats，infrastructure)打印信息。
```
Comment (QC-L) [inline src/content/api/plugins.md:138]:
```suggestion
日志的 API 在 webpack 4.37 版本后提供支持。当 `logging` 在 [`统计配置(stats configuration)`](/configuration/stats/#statslogging)中可用和(或)当 [`infrastructure logging`](/configuration/other-options/#infrastructurelogging) 可用的时候，插件会通过各自的记录格式(stats，infrastructure)打印信息。
```

File: src/content/api/plugins.md [hunk 18]
```diff
@@ -11,102 +11,102 @@ contributors:
...
-This small library is a core utility in webpack but can also be used elsewhere
-to provide a similar plugin interface. Many objects in webpack extend the
-`Tapable` class. The class exposes `tap`, `tapAsync`, and `tapPromise` methods
-which plugins can use to inject custom build steps that will be fired
-throughout a compilation.
+这个小型库是 webpack 的一个核心工具，但也可用于其他地方，
+以提供类似的插件接口。
+在 webpack 中的许多对象都扩展自 `Tapable` 类。
+它对外暴露了 `tap`，`tapAsync` 和 `tapPromise` 等方法，
+插件可以使用这些方法注入到自定义的构建步骤中，这些步骤将在整个编译过程的不同时机触发。
```
Comment (QC-L) [inline src/content/api/plugins.md:33]:
```suggestion
插件可以使用这些方法向 webpack 中注入自定义构建的步骤，这些步骤将在构建过程中触发。
```

Comment (QC-L) [reply src/content/api/plugins.md:33]: 感觉这样更容易理解些~

**Comment by chenzesam** [comment]
@QC-L 都合并完了

**Pull request merged**